libraries/ms-common/src/stm32f0xx/soft_timer.c

```diff
@@ -0,0 +1,105 @@
+#include "soft_timer.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#include "status.h"
+#include "stm32f0xx_rcc.h"
+#include "stm32f0xx_tim.h"
+
+typedef struct timer {
+  uint32_t duration;
+  uint32_t time_started;
+  uint32_t last_update_value;
+  void *context;
+  bool inuse;
+  bool rolled_over;
+  SoftTimerCallback callback;
+} timer;
+
+static timer s_soft_timer_array[SOFT_TIMER_MAX_TIMERS];
+
+// Used to track rollovers this will happen every ~1.19 hrs as we are using microseconds.
+
+void soft_timer_init(void) {
+  for (uint32_t i = 0; i < SOFT_TIMER_MAX_TIMERS; i++) {
+    s_soft_timer_array[i].inuse = false;
+    s_soft_timer_array[i].last_update_value = 0;
+  }
+
+  RCC_APB1PeriphClockCmd(RCC_APB1Periph_TIM2, ENABLE);
+
+  RCC_ClocksTypeDef clock_speeds;
+  RCC_GetClocksFreq(&clock_speeds);
+
+  // Configure each clock tick to be 1 microsecond from 0 to UINT32_MAX. Uses the external 8 MHz
+  // clock for tick source. This will need to adjusted or made into a variable if we choose a
+  // different external clock from the discovery board.
+  TIM_TimeBaseInitTypeDef init_struct = {.TIM_Prescaler = clock_speeds.SYSCLK_Frequency / 1000000,
```

Comment:
I think our coding style guide requires a space between ``{`` and ``.TIM_Prescaler``